### ISSUE TITLE
False positive no-unreachable catch in generator 

### ISSUE THREAD

**Issue by etienne-dldc** [eslint/eslint#12880]
The `no-unreachable` rule does not understand that `yield` can throw an error an report an error.

Here is small repro:

```js
/*eslint no-unreachable: "error"*/

function* generator() {
  try {
    yield 1;
    yield 2;
    return 3;
  } catch (err) { // eslint warn about the catch being unreachable
    return err;
  }
}

var gen = generator();
gen.throw(5) // this will run the code in the catch
```

[Online demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdW5yZWFjaGFibGU6IFwiZXJyb3JcIiovXG5cbmZ1bmN0aW9uKiBnZW5lcmF0b3IoKSB7XG4gIHRyeSB7XG4gICAgeWllbGQgMTtcbiAgICB5aWVsZCAyO1xuICAgIHJldHVybiAzO1xuICB9IGNhdGNoIChlcnIpIHtcbiAgICByZXR1cm4gZXJyO1xuICB9XG59XG5cbnZhciBnZW4gPSBnZW5lcmF0b3IoKTtcbi8vIHJlYWNoIHRoZSB1bnJlYWNoYWJsZSA/XG5nZW4udGhyb3coNSlcbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7fSwiZW52Ijp7ImJyb3dzZXIiOmZhbHNlfX19)

**Comment by kaicataldo** [comment]
Thanks for reporting this. Agreed that this is a bug. Documentation describing this behavior can be found [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator/throw).
